filter.go

```diff
@@ -173,10 +212,13 @@ func (m Middleware) ServeHTTP(w http.ResponseWriter, r *http.Request, next caddy
 	csr.FlushHeaders()
 	if m.compiledContentTypeRegex.MatchString(csr.Recorder().Result().Header.Get("Content-Type")) {
 		buf := new(bytes.Buffer)
+		repl := r.Context().Value(caddy.ReplacerCtxKey).(*caddy.Replacer)
 		if _, err := buf.ReadFrom(csr.Recorder().Result().Body); err != nil {
 			return fmt.Errorf("failed to read from response body: %w", err)
 		}
-		replaced := m.compiledSearchRegex.ReplaceAll(buf.Bytes(), []byte(m.Replacement))
+		replaced := m.compiledSearchRegex.ReplaceAllFunc(buf.Bytes(), func(input []byte) []byte {
```

Comment:
Perhaps I missed something, but is it possible to generate the replacement string ahead of time?

For example:

```go
// When m.Replacement is "abc{system.os}def"
realReplacement := paramReplacementPattern.ReplaceAllFunc(replacement, func(input2 []byte) []byte {
		return m.paramReplacer(input2, groups, repl)
	})
// realReplacement is "abclinuxdef"
replaced := m.compiledSearchRegex.ReplaceAll(buf.Bytes(), realReplacement)
```

Reply:
That makes no sense, as the caddy placeholders also contain request-specific, reverse-proxy-specific, ... placeholders. If we generate the replacement string ahead of time (I think you mean inside `Provision`?), than we won't be able to use those placeholders. And thats the usecase that I need. The code is btw similar to the one of caddy v1 filter module.

Reply:
I see. Thanks for explaining this